### ISSUE TITLE
Incorrect inline-block flow

### ISSUE THREAD

**Issue by Spreeuw** [dompdf/dompdf#2755]
It looks like there's an error in the width calculations of `inline-block` elements:

HTML: <details>

```html
<style type="text/css">
	ul {
		display: inline-block;
		list-style: none;
		padding: 0;
		margin-bottom: 0;
	}

	ul li {
		display: inline-block;
		padding-left: 3px;
		padding-right: 3px;
		margin-bottom: 0;
	}


	ul li strong {
		display: inline-block;
		font-weight: normal;
	}

	ul li p {
		margin: 0;
		padding: 0;
		display: inline-block;
	}
</style>
<ul><li><strong>F&uuml;llmenge:</strong> <p>500g</p></li></ul>
```
</details>

![fullmenge-dev-master](https://user-images.githubusercontent.com/3217773/150517361-d1609009-4ad9-4382-91ca-80c8db2f03f1.png)

In 1.0.2 it looked like this:
![fullmenge-1 0 2](https://user-images.githubusercontent.com/3217773/150517405-0b1d23f8-1a1d-4c48-8f2d-4ecf6ac09a8c.png)

**Comment by Mellthas** [comment]
I need to set the font to DejaVu Sans, cannot reproduce with the default font.

**Comment by Mellthas** [comment]
I think this is the same issue as #2560, and the problem is that float values are compared naively, not accounting for inaccuracies with float computation. The shrink-to-fit width is calculated first, then the inline frames are laid out, checking the available space left in each step; so the numbers being added up are not exactly the same, which can result in the values in the end being marginally off. For inline blocks, this was not an issue prior to version 1.1.0, since they didn’t use proper shrink-to-fit width. Only column widths in tables were affected.

I had a look at this a while ago, but didn’t pursue it further. Maybe we can resolve it now, will have a look in the coming days.

**Comment by Mellthas** [comment]
Thanks for reporting these issue, by the way!

**Comment by Spreeuw** [comment]
> I need to set the font to DejaVu Sans, cannot reproduce with the default font.

Sorry about not doing the example/testing with the default serif font. I think you are right about this being dependent on rounding in specific cases. After playing with the font size a bit I was able to reproduce with the default serif font too, at `21pt`, adding this to the example above:
```css
	body {
		font-family: serif;
		font-size: 21pt;
	}
```

**Comment by Mellthas** [comment]
Yes, that also works for me. Small differences in numbers can make the issue appear and disappear.

Oh, and just realized, I meant _Thanks for reporting these **issues**, by the way!_ (and helping with testing the pre-release).

**Comment by Spreeuw** [comment]
> Oh, and just realized, I meant Thanks for reporting these issues, by the way! (and helping with testing the pre-release).

I think I am the one who should think _you_ for all the great work you have done so far on this release, it looks like you've been very busy cleaning up old code 💪

**Comment by Mellthas** [comment]
I am going to close this as a duplicate of #2560, as it is fixed by #2761, too.
